src/Microsoft.ML.Data/Data/RowCursorUtils.cs

```diff
@@ -325,30 +325,6 @@ private static Func<bool> GetIsNewGroupDelegateCore<T>(Row cursor, int col)
             };
         }
 
-        [Obsolete("The usages of this appear to be based on a total misunderstanding of what Batch actually is. It is a mechanism " +
-            "to enable sharding and recovery of parallelized data, and has nothing to do with actual data.")]
-        [BestFriend]
-        internal static Func<bool> GetIsNewBatchDelegate(Row cursor, int batchSize)
```

Comment:
>GetIsNewBatchDelegate [](start = 35, length = 21)

This is used in our internal repo. it is used for scoring models that score a whole batch of examples at a time. It may be useful here in the future as well, for example, TF has the capability to score multiple examples at a time as well (may give speedups when using GPUs). 

By the comment on the ObsoleteAttribute above, it seems that the name of this method might be misleading - perhaps we could keep the method but find a more appropriate name for it?

Reply:
Thanks, @yaeldekel! Shouldn’t this be in the internal repo then? Otherwise this is just dead code sitting here with an assumption it may be used in future.

Reply:
Hi @yaeldekel and @codemzs, I wrote this obsolete comment. Whoever wrote this method, as I said, completely misunderstood what `Batch` on a row cursor was for. Using `Batch` in this fashion is completely pointless -- the batch is meant as a reconcilation mechanism for parallel cursors, as described in [its documentation](https://github.com/dotnet/machinelearning/blob/faffd179c961f120ec4e6babb06bbfb2cca6a6ea/src/Microsoft.ML.Core/Data/IDataView.cs#L169), further [code documentation](https://github.com/dotnet/machinelearning/blob/master/src/Microsoft.ML.Core/Data/ICursor.md#batch), literally every single usage of `Batch` throughout the codebase.

@codemzs's removal of it is appropriate. The internal code may be using it, but, as explained, that's because someone committed a blunder.